File: server/src/main/java/com/smartexpense/api/domain/repository/NotificationRepository.java
package com.smartexpense.api.domain.repository;

import com.smartexpense.api.domain.model.Notification;
import java.util.List;
import java.util.Optional;
import java.util.UUID;

public interface NotificationRepository {
    List<Notification> findAllByUserIdOrderByCreatedAtDesc(UUID userId);
    Optional<Notification> findByIdAndUserId(UUID id, UUID userId);
    Notification save(Notification notification);
    void markAllAsReadByUserId(UUID userId);
}

File: server/src/main/kotlin/com/smartexpense/api/domain/repository/NotificationRepository.kt
package com.smartexpense.api.domain.repository

import com.smartexpense.api.domain.model.Notification
import java.util.UUID

interface NotificationRepository {
    fun findAllByUserIdOrderByCreatedAtDesc(userId: UUID): List<Notification>
    fun findByIdAndUserId(id: UUID, userId: UUID): Notification?
    fun save(notification: Notification): Notification
    fun markAllAsReadByUserId(userId: UUID)
}
